mypyc/lib-rt/bytes_ops.c

```diff
@@ -5,6 +5,27 @@
 #include <Python.h>
 #include "CPy.h"
 
+CPyTagged CPyBytes_GetItem(PyObject *o, CPyTagged index) {
+    if (CPyTagged_CheckShort(index)) {
+        Py_ssize_t n = CPyTagged_ShortAsSsize_t(index);
+        Py_ssize_t size = ((PyVarObject *)o)->ob_size;
+        if ((n >= 0 && n >= size) || (n < 0 && n + size < 0)) {
+            PyErr_SetString(PyExc_IndexError, "index out of range");
+            return CPY_INT_TAG;
+        }
+        if (n < 0)
```

Comment:
You can possibly avoid a comparison by only checking for `n < 0` once. You can have a fast path for non-negative values in range, and handle everything else in a slow path. For example, you could have something like this:

```
if (n < 0 || n >= size) {
    // Slow path
    if (likely(n < 0 && n + size >= 0)) {
        n += size;
    } else {
        // raise exception
    }
}
```

It's unclear whether this will be faster than what you have, since C compilers are pretty smart, but at least it makes the fast and slow paths clearer for the reader.